packages/integrations/cloudflare/src/index.ts

```diff
@@ -136,7 +149,7 @@ export default function createIntegration(args?: Options): AstroIntegration {
 					await fs.promises.mkdir(functionsUrl, { recursive: true });
 				}
 
-				if (isModeDirectory && _buildConfig.split) {
+				if (isModeDirectory && (_buildConfig.split || functionPerRoute)) {
```

Comment:
We could remove the check for `_buildConfig.split`

Reply:
Not yet, it should stay backward compatible.